### PR TITLE
feat: refined ux update a taxonomy by downloading and uploading  (TEMP)

### PR DESCRIPTION
**Pull request by rpenido**
## Description
This PR improves the import tags functionality for existing taxonomies implemented at https://github.com/openedx/frontend-app-course-authoring/pull/675.

## Supporting Information
* Closes:
  * https://github.com/openedx/modular-learning/issues/126
* Depends on:
  * https://github.com/openedx/frontend-app-course-authoring/pull/675
  * https://github.com/openedx/openedx-learning/pull/130

### Re-import tags to an existing taxonomy
1. Have an import taxonomy template ready to be imported. You could get one from below:
    - [template.csv](https://github.com/openedx/frontend-app-course-authoring/files/13322469/template.csv)
    - [template.json](https://github.com/openedx/frontend-app-course-authoring/files/13322474/template.json)
1. Go to the taxonomy list page (http://localhost:2001/taxonomies)
1. Click on the `Re-import` menu from an existing taxonomy card
1. You will be prompted with the first page, where you can export the current taxonomy in CVS or JSON format. Check if both buttons export the taxonomy in the desired format:
image
1. Click on `Next`
1.  Will you have to select a CSV or JSON format file to import. You could do this clicking on the component or using drag and drop. Only .csv and .json files should be accepted.
image
1. Click on `Import` to go to the next step.
1. This page will show the preview actions for this import. Check it and click on `Continue`. If the file shows no differences, the `Continue` button will be disabled. 
image
1. A new confirmation windows will be show. Clicking on `Yes, import file` should import the tags, replacing the old ones.
image
1. The Tag List should be automatically updated.

____
Private-ref:
 - [FAL-3539](https://tasks.opencraft.com/browse/FAL-3539)

### CODE REVIEW

File: src/taxonomy/import-tags/ImportTagsWizard.jsx [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+import {
+  DeleteOutline,
+  Download,
+  InsertDriveFile,
+  Warning,
+} from '@edx/paragon/icons';
+import PropTypes from 'prop-types';
+import { useState } from 'react';
+
+import { getFileSizeToClosestByte } from '../../files-and-videos/generic/utils'; // ToDo: Check best approach
```
Comment (ChrisChV) [inline src/taxonomy/import-tags/ImportTagsWizard.jsx]: @rpenido Maybe you need to extract this to `src/utils`

File: src/taxonomy/import-tags/data/api.js [hunk 2]
```diff
@@ -40,19 +50,62 @@ export async function importNewTaxonomy(taxonomyName, taxonomyDescription, file)
...
+      formData.append('file', file);
+
+      await getAuthenticatedHttpClient().put(
+        getTagsImportApiUrl(taxonomyId),
+        formData,
+      );
+    },
+    onSuccess: (_data, variables) => {
+      queryClient.invalidateQueries({
+        queryKey: ['tagList', variables.taxonomyId],
```
Comment (ChrisChV) [inline src/taxonomy/import-tags/data/api.js:86]: @rpenido I think also you need to invalidate the `taxonomyDetail` query

Comment (rpenido) [reply src/taxonomy/import-tags/data/api.js:86]: Thank you @ChrisChV! I missed the tag count on Taxonomy.
Done 111904180de7951ec038b272cae5711c62ce8962

File: src/taxonomy/import-tags/ImportTagsWizard.jsx (hunk 1 above)
Comment (rpenido) [reply src/taxonomy/import-tags/ImportTagsWizard.jsx]: Done 6a55c6f3c6895a9b5b96b1f8ae9e30c3fdff06c

File: src/taxonomy/import-tags/data/api.test.js [hunk 3]
```diff
@@ -1,48 +0,0 @@
-import MockAdapter from 'axios-mock-adapter';
```
Comment (rpenido) [inline src/taxonomy/import-tags/data/api.test.js:1]: This was renamed to `.jsx` to allow <QueryClientProvider> component wrapper

File: src/taxonomy/import-tags/data/utils.js [hunk 4]
```diff
@@ -89,32 +90,3 @@ export const importTaxonomy = async (intl) => {
       console.error(error.response);
     });
 };
-
```
Comment (rpenido) [inline src/taxonomy/import-tags/data/utils.js:92]: This function was replaced by the ImportTagsWizard

File: src/files-and-videos/generic/utils.js [hunk 5]
```diff
@@ -1,23 +1,4 @@
-export const getFileSizeToClosestByte = (fileSize, numberOfDivides = 0) => {
```
Comment (rpenido) [inline src/files-and-videos/generic/utils.js:1]: This function was moved to /src/utils.js

**Comment by ChrisChV** [comment]
@rpenido I can't continue with the import. Maybe I'm missing something?

https://www.loom.com/share/4bf7b266cda4407f914b7100a39fe639?sid=ff2bc3b2-78b1-4c8a-b088-1ba793a8a77b

File: src/taxonomy/import-tags/ImportTagsWizard.jsx [hunk 6]
```diff
@@ -0,0 +1,374 @@
...
+  };
+
+  return (
+    <Stepper.Step eventKey="upload" hasError={importPlanError}>
+      <Stack gap={3} data-testid="upload-step">
+        <p>{intl.formatMessage(messages.importWizardStepUploadBody, { br: linebreak })}</p>
+        <div>
+          {!file ? (
+            <Dropzone
+              style={{ height: '200px' }}
```
Comment (ChrisChV) [inline src/taxonomy/import-tags/ImportTagsWizard.jsx]: @rpenido It seems like you are using this style in several places. maybe it's better to [create a utility](https://github.com/openedx/frontend-app-course-authoring/commit/dab9d8a01a0389826e989252b112c3b70869e251#diff-c2d8998567caf1b28fc98e349d9fdbe0b95a3e4ea15938b79b7ce147b90b9cbdR5). Ref: https://github.com/openedx/frontend-app-course-authoring/pull/723#discussion_r1425040465

**Comment by ChrisChV** [comment]
Also we need to improve the error messages here, but it can be on a follow-up task CC @bradenmacdonald 

image

**Comment by rpenido** [comment]
> @rpenido I can't continue with the import. Maybe I'm missing something?
> 
> https://www.loom.com/share/4bf7b266cda4407f914b7100a39fe639?sid=ff2bc3b2-78b1-4c8a-b088-1ba793a8a77b

Thank you, @ChrisChV! I fixed a bug in the error alert message here: ad69e11344172cc313f91c16060af76e8e84f806

But I think you also found a bug in our import function. The step `#30` is showing this:
```
[2023-12-18 16:51:19]  #30: Delete tag (external_id=None) [Started]
[2023-12-18 16:51:19] MultipleObjectsReturned('get() returned more than one Tag -- it returned more than 20!') 
```

Edit:
https://github.com/openedx/openedx-learning/blob/3e2d6a497efddfae7a1c24b361ddfd36704dbb80/openedx_tagging/core/tagging/import_export/import_plan.py#L137-L140

We rely on `external_id` to check if some Tag in our DB is being imported. But we also delete the Tags using the `external_id`. The tags for this taxonomy don't have the `external_id` (this can occur, right?). This is causing the error above.

Should I handle it or create a new task for this?

File: src/taxonomy/import-tags/ImportTagsWizard.jsx (hunk 6 above)
Comment (rpenido) [reply src/taxonomy/import-tags/ImportTagsWizard.jsx]: Thank you! FIxed here: 2d3abe8542acc16d49bbceb59aed9e1fe82e5bd7

**Comment by ChrisChV** [comment]
> Should I handle it or create a new task for this?

@rpenido  I think we need to do it now to get it operational for the sandbox. I can fix it if you want, I have time for it

**Comment by rpenido** [comment]
> I can fix it if you want, I have time for it

I would appreciate that @ChrisChV !

If this is blocking for testing here, the current code will work reimporting any taxonomy created via Import.

**Comment by ChrisChV** [comment]
> > I can fix it if you want, I have time for it
> 
> I would appreciate that @ChrisChV !
> 
> If this is blocking for testing here, the current code will work reimporting any taxonomy created via Import.

@rpenido I fixed this bug on this PR: https://github.com/openedx/openedx-learning/pull/135
I need to add tests, but you can review it and test it with your PR

**Comment by rpenido** [comment]
> @rpenido I fixed this bug on this PR: [openedx/openedx-learning#135](https://github.com/openedx/openedx-learning/pull/135) I need to add tests, but you can review it and test it with your PR

I will check it tomorrow. Thank you @ChrisChV !

**Review by ChrisChV — APPROVED** [review]
@rpenido :+1:
- [X] I tested this: I followed the testing instructions
- [X] I read through the code and considered the security, stability and performance implications of the changes.
- [X] I tested that the UI can be used with a keyboard only (tab order, keyboard controls).
- [X] Includes tests for bugfixes and/or features added.
- [X] Includes documentation

**Comment by rpenido** [comment]
Closed in favor of https://github.com/openedx/frontend-app-course-authoring/pull/732

**Pull request closed without merging**